framework/framework.go

```diff
@@ -97,13 +98,43 @@ func (f *Framework) ProcessCreate(obj interface{}, resources ...Resource) error
 	return nil
 }
 
+// ProcessCreateWithBackoff is the same as ProcessCreate but takes an additional
+// backoff factory returning new backoff implementations to retry the creation
+// on resource errors. Retries are implemented using a retry resource that wraps
+// each given resource. Errors are logged eventually in case an appropriate
+// logger is configured in the operator framework.
+func (f *Framework) ProcessCreateWithBackoff(obj interface{}, resources []Resource, backoffFactory func() backoff.BackOff) error {
+	var retryResources []Resource
+
+	for _, r := range resources {
+		resourceConfig := DefaultRetryResourceConfig()
+		resourceConfig.BackOff = backoffFactory()
+		resourceConfig.Logger = f.logger
+		resourceConfig.Resource = r
+
+		retryResource, err := NewRetryResource(resourceConfig)
```

Comment:
Thanks for your feedback. I want to keep it that way because it shows how composable the resources are. For me this a evidence of good design. Adding the retry logic hard coded into the Process* functions feels to me like a hack. Tomorrow something else has to be added and it gets even more dirty. Instead we can then simply add another resource that does the things it should and everything is nicely organized, separated, tested and still customizable.